modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/util/AssetPublisherHelperImpl.java

```diff
@@ -1108,6 +1112,25 @@ protected void activate(Map<String, Object> properties)
 		return assetEntryResults;
 	}
 
+	private long[] _getSegmentsEntryIds(PermissionChecker permissionChecker) {
+		ServiceContext serviceContext =
+			ServiceContextThreadLocal.getServiceContext();
+
+		if (serviceContext == null) {
+			return new long[0];
+		}
+
+		HttpServletRequest httpServletRequest = serviceContext.getRequest();
```

Comment:
Would it be better to extract the `HttpServletRequest` from the portlet request? Something like

````
_portal.getHttpServletRequest(portletRequest);

```

Reply:
Good catch, thank you :)

Reply:
Change it!!